### PR TITLE
[NamedOptimizer][2/N] Prepare the enablement of state_dict for FSDP

### PR DESCRIPTION
**Pull request by fduwjj**
1. Add param_group check logic and unit test
2. Remove unnecessary check for conditional param update
3. Return the param_group from the inner optimizer so that when param_group is None or not all params are specified, we still return the expected result. 

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #91147

### CODE REVIEW

File: torch/distributed/optim/named_optimizer.py [hunk 1]
```diff
@@ -171,9 +190,7 @@ def load_state_dict(self, state_dict: Mapping[str, Any]) -> None:
             )
         for idx, param_key in enumerate(self.ordered_param_keys):
             if param_key not in state.keys():
-                raise ValueError(
```
Comment (fegin) [inline torch/distributed/optim/named_optimizer.py:174]: Can we add a comment to describe why this is legit?

**Review by fegin — APPROVED** [review]
Left a comment. Accept to unblock.

File: torch/distributed/optim/named_optimizer.py (hunk 1 above)
Comment (fduwjj) [reply torch/distributed/optim/named_optimizer.py:174]: Sure. Let me explain it with comments.

**Comment by fduwjj** [comment]
@pytochbot merge

**Comment by fduwjj** [comment]
@pytochbot merge

File: torch/distributed/optim/named_optimizer.py [hunk 2]
```diff
@@ -170,10 +189,9 @@ def load_state_dict(self, state_dict: Mapping[str, Any]) -> None:
                 f"Expects equal length as {len(new_state)} in `state_dict` state length but found {len(state)}."
             )
         for idx, param_key in enumerate(self.ordered_param_keys):
+            # When the conditional training is performed, not all parameters are updated in the optim.
```
Comment (rohan-varma) [inline torch/distributed/optim/named_optimizer.py:192]: Can we add a parity test with regular optimizer under conditional training use case?

Comment (fduwjj) [reply torch/distributed/optim/named_optimizer.py:192]: Done, added in https://github.com/pytorch/pytorch/pull/91344

**Pull request closed without merging**